### PR TITLE
feat: map custom app rating api 

### PR DESCRIPTION
**Pull request by AndrewAminInstabug**
iOS: Instabug.willRedirectToAppStore()

Android: Instabug.willRedirectToAppStore();

This API is mainly used by the customer to mark that the user was redirected from the custom prompt to the App store or Play store so we can try to match the review to the relevant session replay.

## Type of change
- [ ] Bug fix (non-breaking change that fixes an issue)
- [x] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

### Development
- [x] Lint rules pass locally
- [x] The code changed/added as part of this pull request has been covered with tests
### Code review 
- [ ] This pull request has a descriptive title and information useful to a reviewer
- [ ] Issue from task tracker has a link to this pull request

### CODE REVIEW

File: lib/src/modules/instabug.dart [hunk 1]
```diff
@@ -412,4 +412,9 @@ class Instabug {
     final darkKey = await dark.obtainKey(configuration);
     return _host.setCustomBrandingImage(lightKey.name, darkKey.name);
   }
+
+  /// This API has to be call when using custom app rating prompt
```
Comment (abdelhamid-f-nasser) [inline lib/src/modules/instabug.dart:416]: How about we improve the documentation a little bit to add more context about the API like:
```suggestion
  /// Allows detection of app review sessions which are submitted through custom prompts.
  ///
  /// Use this when utilizing a custom app rating prompt. It should be called
  /// once the user clicks on the Call to Action (CTA) that redirects them to the app store.
  /// Helps track session data for insights on user interactions during review submission.
```

File: lib/src/modules/instabug.dart [hunk 2]
```diff
@@ -412,4 +412,9 @@ class Instabug {
...
+  static Future<void> willRedirectToAppStore() async {
```
Comment (abdelhamid-f-nasser) [inline lib/src/modules/instabug.dart:417]: What are your thoughts about renaming it to a more general name that fits all platforms/stores like `willRedirectToStore` or `willRedirectToPlatformStore`

Please share your thoughts @Instabug/cross-platform-team.

**Review by abdelhamid-f-nasser — CHANGES_REQUESTED** [review]
Great effort in mapping your first API 🎉 💪, please check the suggestions.

File: lib/src/modules/instabug.dart (hunk 2 above)
Comment (a7medev) [reply lib/src/modules/instabug.dart:417]: I second that. `willRedirectToStore` feels more platform-agnostic to me.

**Review by abdelhamid-f-nasser — APPROVED** [review]

**Pull request merged**